src/Facade/Services/ImportBookReportFacadeService.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Linn.Stores.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Reporting.Models;
+    using Linn.Common.Reporting.Resources.Extensions;
+    using Linn.Stores.Domain.LinnApps.Reports;
+    using Linn.Stores.Resources.ImportBooks;
+
+    public class ImportBookReportFacadeService : IImportBookReportFacadeService
+    {
+        private readonly IImportBookReportService reportService;
+
+        public ImportBookReportFacadeService(IImportBookReportService reportService)
+        {
+            this.reportService = reportService;
+        }
+
+        public IResult<ResultsModel> GetImpbookIPRReport(IPRSearchResource resource)
+        {
+            DateTime from;
+            DateTime to;
+            try
+            {
+                from = DateTime.Parse(resource.FromDate);
+                to = DateTime.Parse(resource.ToDate);
+            }
+            catch (Exception)
+            {
+                return new BadRequestResult<ResultsModel>("Invalid dates supplied to impbook IPR report");
+            }
+
+            return new SuccessResult<ResultsModel>(this.reportService.GetIPRReport(from, to));
+        }
+
+        public IResult<IEnumerable<IEnumerable<string>>> GetImpbookIprReportExport(IPRSearchResource resource)
+        {
+            DateTime from;
+            DateTime to;
```

Comment:
if you do go down the TryParse route you could ditch these declarations and just do

`DateTime.TryParse(dateString, out var from)`

and same for to

Reply:
I really like how clean this is, but just writing a test to see what it does and it seems to pass in "01/01/0001" when fed nonsense. Are we ok with that cause it's basically not going to get called without the right values? I guess worst case is just that it could do a slow select which doesn't really matter

Reply:
catch the bool it returns and you'll know if it was valid or not